### ISSUE TITLE
Cannot assume a role via a regional STS endpont for a provider in a different region

### ISSUE THREAD

**Issue by ctavplay** [hashicorp/terraform-provider-aws#23189]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```bash
$ terraform -v
Terraform v1.1.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.0.0
```

### Affected Resource(s)

* aws provider

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.0.0"
    }
  }
}

provider "aws" {
  alias = "ap-northeast-1"
  region = "ap-northeast-1"

  assume_role {
    role_arn = "arn:aws:iam::XXXXXXXX:role/some_role"
  }

  endpoints {
    sts = "https://sts.us-east-1.amazonaws.com"
  }
}

data "aws_availability_zones" "zones" {
  provider = aws.ap-northeast-1
}
```

### Debug Output

This is the relevant excerpt from the debug log:

```
2022-02-14T18:17:10.840-0500 [DEBUG] provider.terraform-provider-aws_v4.0.0_x5: [aws-sdk-go-v2] Request
POST / HTTP/1.1
Host: sts.us-east-1.amazonaws.com
User-Agent: APN/1.0 HashiCorp/1.0 Terraform/1.1.3 (+https://www.terraform.io) terraform-provider-aws/4.0.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go-v2/1.13.0 os/macos lang/go/1.17.6 md/GOOS/darwin md/GOARCH/arm64 api/sts/1.14.0
Content-Length: 146
Amz-Sdk-Invocation-Id: 1c6aa1b6-4df6-42b3-a985-cc348dd37551
Amz-Sdk-Request: attempt=1; max=1
Authorization: AWS4-HMAC-SHA256 Credential=[REDACTED]/20220214/ap-northeast-1/sts/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=[REDACTED]
Content-Type: application/x-www-form-urlencoded
X-Amz-Date: 20220214T231710Z
X-Amz-Security-Token: [REDACTED]
Accept-Encoding: gzip

Action=[REDACTED]
2022-02-14T18:17:10.977-0500 [DEBUG] provider.terraform-provider-aws_v4.0.0_x5: [aws-sdk-go-v2] Response
HTTP/1.1 403 Forbidden
Connection: close
Content-Length: 322
Content-Type: text/xml
Date: Mon, 14 Feb 2022 23:17:10 GMT
Keep-Alive: timeout=5
X-Amzn-Requestid: dec71d3e-32ef-4f43-8f0e-568de4f10526
```

### Expected Behavior

I expected this to work as it does with provider version 3.65.0.

### Actual Behavior

I get this error:
```
Error: operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 4473c800-ee14-466d-bf5a-6b36f5617b7e, api error SignatureDoesNotMatch: Credential should be scoped to a valid region, not 'ap-northeast-1'.
```

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

### Important Factoids

I tried this same code with provider version 3.65.0 and it worked.  The difference between the two is the Authorization header in the POST to the us-east-1 STS endpoint, the region specified under "Credential" is ap-northeast-1 under 4.0.0 and not us-east-1 as done in 3.65.0.

This code runs in a VPC in us-east-1 and all role assumptions must go through the local VPC endpoint.  We need to be able to deploy resources in other regions, however.

**Comment by devmohammedothman** [comment]
I have similar issue with this error

**Comment by Doug-North** [comment]
I have a similar issue with the common denominator being "Error: no EC2 IMDS role found, operation error ec2imds: GetMetadata, canceled, context deadline exceede"

This inhibits our ability to use AWS Provider version 4.0.0, previous versions work just fine  and I can retrieve token manually on the ec2 instance.

**Comment by gdavison** [comment]
Hello all, starting in version 4.2, to use an alternate region for STS, use the new parameter `sts_region` instead of specifying an alternate STS endpoint.

This was unfortunately missed from the initial releases, and has been merged in #23212 and will be released later this week
